lumapu/ahoy
# Control

Hi,

TESTET for HM-1500 (3rd gen)

- devControl commands via MQTT Topic (<YOUR TOPIC>/devcontrol/subcmd + payload); eg. mysolar/devcontrol/1/0 --> turn on inverter 1; eg. mysolar/devcontrol/1/1 --> turn off; mysolar/devcontrol/0/11 + payload "300" --> set power limit for inverter 1 to 300 W (set power is now remanent during inverter power cycle and remanent during power cycle of ahoy and power limit is set on each reboot/start up THX: @klahus1)
- Set power limit in setup page; Save --> writes to eeprom --> reboot --> after reboot set power limit acc. to eeprom
- Set power limit to zero --> inverter will not stop working (Reactive Power <> 0 VAr) but active power will be zero
- During change of power limit eg. 200 W --> 500 W, the Powerfactor is not controled
- Case "set to unlimited power": NOT working by setting it to large numbers or -1 or similar. You have to set it acc. to the specific device eg. Mi-1500 --> 1500. There is an error handling because the response from inverter differs in case the new power limit is NOT accepted. See logs.

Open:

- prototype for a Web Interface "/devcontrol" created. NO content / function yet.

Suggestion @maintainers:
- create a "dev" branch and merge / accept against this branch. Collect feedback in issues. After the community build/tested then merge against main (= release)
--> "_aschiffler wants to merge 8 commits into grindylow:development_control_"
